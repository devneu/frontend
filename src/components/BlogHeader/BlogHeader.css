.blog-header{
   text-align: center;
}
.blog-header__title{
   margin: 0
}
.blog-header__subtitle{
   font-weight: 300;
}