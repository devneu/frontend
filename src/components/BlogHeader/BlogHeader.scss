.blog-header {
  text-align: center;
  background-color: #f8f9fa;
  margin-bottom: 20px;
}
