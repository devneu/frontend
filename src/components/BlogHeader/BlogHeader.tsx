import React from 'react';
import { Typography, PageHeader } from 'antd';
import './BlogHeader.scss'

const { Title } = Typography;

interface BlogHeaderProps {
   title: string,
   subtitle: string
}
const BlogHeader = ({ title, subtitle }: BlogHeaderProps) => {
   return (
      <PageHeader
         className="blog-header">
         <Title
            level={1}
            className="blog-header__title">{title}</Title>
         <Title
            level={3}
            type="secondary"
            className="blog-header__subtitle">{subtitle}</Title>
      </PageHeader>
   )
}

export default BlogHeader;