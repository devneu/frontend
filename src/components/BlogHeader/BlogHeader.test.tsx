import { render } from '@testing-library/react';
import BlogHeader from './BlogHeader';
import { BlogHeaderProps } from './BlogHeader'
import renderer from 'react-test-renderer';
// const setUp = (props: BlogHeaderProps) => render(<BlogHeader {...props} />);
// const title = "Blog Title";
// const subtitle = "Blog Subtitle";

// describe("should render BlogHeader Component", () => {
//    let component: any;
//    beforeEach(() => {
//       component = setUp({ title: title, subtitle: subtitle });
//    })
//    it("should contain title with text from props", () => {
//       const componentTitle = component.getByText(title);
//       expect(componentTitle.innerHTML).toBe(title);
//    })
//    it("should contain subtitle with text from props", () => {
//       const componentTitle = component.getByText(subtitle);
//       expect(componentTitle.innerHTML).toBe(subtitle);
//    })
// })



describe('BlogHeader', () => {
   it('Should successfully render', () => {
      const wrapper = renderer
         .create(
            <BlogHeader title="Blog Title" subtitle="Blog Subtitle" />
         )
         .toJSON();
      expect(wrapper).toMatchSnapshot();
   });
});