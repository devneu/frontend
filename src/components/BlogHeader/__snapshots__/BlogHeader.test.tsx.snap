// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogHeader Should successfully render 1`] = `
<div
  className="ant-page-header blog-header ant-page-header-ghost"
>
  <div
    className="ant-page-header-content"
  >
    <h1
      className="ant-typography blog-header__title"
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Blog Title
    </h1>
    <h3
      className="ant-typography ant-typography-secondary blog-header__subtitle"
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Blog Subtitle
    </h3>
  </div>
</div>
`;
